submissions/Dovahkiin1991/friends-app/app.js

```diff
@@ -0,0 +1,339 @@
+
+//const ITEMS_PER_PAGE = 10; feature pagination
+const usersList = [];
+let usersFilteredList = [];
+const nameInput = document.querySelector('.filter__block input');
+
+const callAPI = async () => {
+    const data = await fetch('https://randomuser.me/api/?results=30&nat=ua');
+    handleErrors(data);
+
+    const { results: users } = await data.json();
+
+    return users;
+};
+
+const handleErrors = (response) => {
+    if (!response.ok) {
+        throw Error(response.statusText);
+    }
+    return response;
+}
+
+const getUsers = async() => {
+    try {
+        const users = await callAPI();
+        usersList.push(...users);
+        renderUsers([]);
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+const getOpt = (name) => {
+    return document.querySelector(`.filter__block input[name="${name}"]:checked`);
+}
+
+const mainContainer = document.getElementById('main__content');
+let usersListContainer = document.createElement('div');
+usersListContainer.classList.add('users');
+
+const renderUsers = (pipes, usersFiltered) => {
+    let users = pipes.reduce((list, func) => {
```

Comment:
Let's make your functions a little closer to the "single responsibility" concept. Your `renderUsers` function doesn't just _renders_ them. It also processes the whole array with filters, sorts, etc. These actions could be separated into another function that receives an array of users, some parameters to process, and returns a new array which then can be used as a basis to build HTML elements and render them to a page.